### PR TITLE
Better Config typing

### PR DESCRIPTION
**Pull request by asazernik**
I'm working on putting less extraneous cruft into ReduxFormContext to avoid unnecessary re-renders. (These were causing bugs in my project.) Since this is a bit of a scary refactor, wanted to make sure (internal-use-only) Flow types were up to snuff first.

### CODE REVIEW

File: src/createReduxForm.js [hunk 1]
```diff
@@ -170,115 +170,120 @@ type OnChangeFunction = (
...
-  onSubmitFail?: OnSubmitFail,
-  onSubmitSuccess?: OnSubmitSuccess,
-  propNamespace?: string,
-  submitAsSideEffect?: boolean,
-  validate?: ValidateFunction,
-  warn?: ValidateFunction,
-  touchOnBlur?: boolean,
-  touchOnChange?: boolean,
-  persistentSubmitErrors?: boolean,
-  registeredFields: any
```
Comment (renatoagds) [inline src/createReduxForm.js]: You forgot to put `registeredFields` back. It's missing in `Config`.

**Comment by renatoagds** [comment]
@asazernik do you have plans to work in the fix?

File: src/createReduxForm.js (hunk 1 above)
Comment (asazernik) [reply src/createReduxForm.js]: It's in Props (in that big literal that comes after `OptionalConfig` in the intersection type chain). As far as I understand it isn't something that a caller passes into `@reduxForm`. Are you seeing any false positives from the typechecker?

**Comment by asazernik** [comment]
Agh! Missed the previous comment. Replied inline.

File: src/createReduxForm.js (hunk 1 above)
Comment (asazernik) [reply src/createReduxForm.js]: (Pushed up a new version with comments describing each type's purpose.)

**Comment by renatoagds** [comment]
Published in [`v8.2.6`](https://github.com/erikras/redux-form/releases/tag/v8.2.6).

**Pull request merged**